### PR TITLE
15770 - Adding company password flow

### PR DESCRIPTION
**Pull request by rodrigo-barraza**
*Issue #:15770*
https://github.com/bcgov/entity/issues/15770

*Description of changes:*
Implementing corporation affiliation processing using company password.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2418-ppojxdbf.web.app

File: auth-web/src/components/auth/manage-business/ManageBusinessDialog.vue [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    &.top-of-list{
+      border-top: 1px solid rgb(228, 228, 228);
+    }
+    .item-content{
+      color: #000 !important;
+    }
+  }
+
+  .v-list-item{
+    background: #FFFFFF;
```
Comment (Jxio) [inline auth-web/src/components/auth/manage-business/ManageBusinessDialog.vue:387]: this can be replaced with `$BCgovInputBG`

**Review by Jxio — COMMENTED** [review]
update package version

File: auth-web/src/components/auth/manage-business/ManageBusinessDialog.vue (hunk 1 above)
Comment (rodrigo-barraza) [reply auth-web/src/components/auth/manage-business/ManageBusinessDialog.vue]: done!

File: auth-web/src/components/auth/manage-business/ManageBusinessDialog.vue [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+                  <div style="color:#313132;">
+                    place holder
+                  </div>
+                </v-list-group>
+
+                <v-list-group v-model="requestAuthRegistryOption">
+                  <template v-slot:activator>
+                    <v-list-item-title>Request authorization from the Business Registry</v-list-item-title>
+                  </template>
+                  <div style="color:#313132;">
```
[the remark this answers is not in the document]
Comment (rodrigo-barraza) [reply auth-web/src/components/auth/manage-business/ManageBusinessDialog.vue]: done!

File: auth-web/src/components/auth/manage-business/ManageBusinessDialog.vue [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+                      To confirm your access, please click on the link in the email. This will add the business to your Business Registry List. The link is valid for 15 minutes.
+                    </div>
+                  </div>
+                </v-list-group>
+
+                <v-list-group v-model="requestAuthBusinessOption">
+                  <template v-slot:activator>
+                    <v-list-item-title>Request authorization from the business</v-list-item-title>
+                  </template>
+                  <div style="color:#313132;">
```
[the remark this answers is not in the document]
Comment (rodrigo-barraza) [reply auth-web/src/components/auth/manage-business/ManageBusinessDialog.vue]: done!

File: auth-web/src/components/auth/manage-business/ManageBusinessDialog.vue [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+                </v-list-group>
+
+                <v-list-group v-model="emailOption">
+                  <template v-slot:activator>
+                    <v-list-item-title>
+                      Confirm authorization using your registered office email address
+                      <div class="subtitle"> (If you forgot or don't have a business {{passwordText}})</div>
+                    </v-list-item-title>
+                  </template>
+                  <div style="color:#313132;">
```
[the remark this answers is not in the document]
Comment (rodrigo-barraza) [reply auth-web/src/components/auth/manage-business/ManageBusinessDialog.vue]: done!

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2418-ppojxdbf.web.app

**Comment by seeker25** [comment]
image

Styling looks off on the right side, also when I try to add a business the box just disappears. This is on my BC Reg staff account.

**Comment by seeker25** [comment]
image

File: auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue [hunk 5]
```diff
@@ -480,12 +480,6 @@ export default class AccountInfo extends Mixins(
     this.$refs.suspensionCompleteDialog.close()
   }
 
-  protected getAccountFromSession (): AccountSettings {
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue:483]: Already in the mixin

File: auth-web/src/components/auth/manage-business/BusinessLookup.vue [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <template>
   <!-- once in Summary state, need to re-mount to reuse this component -->
-  <div id="business-lookup" v-if="state !== States.SUMMARY">
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/BusinessLookup.vue:3]: Removed this so it wont need a remount, i'll look at this more in the next ticket

File: auth-web/src/components/auth/manage-business/SearchBusinessNameRequest.vue [hunk 7]
```diff
@@ -95,13 +124,13 @@ export default class SearchBusinessNameRequest extends Vue {
   @Prop({ default: '' }) readonly userLastName: string
 
   // local variables
-  manageBusinessDialog = { show: false }
+  dialogType = ''
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/SearchBusinessNameRequest.vue:127]: will remove

**Review by seeker25 — APPROVED** [review]

**Comment by seeker25** [comment]
@Jxio @rodrigo-barraza need one of you to code review this, then we can merge

**Review by Jxio — APPROVED** [review]
Looks good to me, thanks Travis!

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2418-ppojxdbf.web.app

**Comment by ethantspitt** [comment]
UXA notes for the modal (using the URL review link ):
Screenshot 2023-08-08 at 8 54 18 AM

**Pull request merged**